src/connection.ts

```diff
@@ -168,6 +184,54 @@ export class Connection extends JSForceConnection {
     this.logger = this.tooling._logger = await Logger.child('connection');
   }
 
+  public async deploy(
+    zipInput: Buffer,
+    options: deployOptions,
+    callback?: Callback<AsyncResult>
+  ): Promise<DeployResultLocator<AsyncResult>> {
+    const rest = options.rest;
+    // neither API expects this option
+    delete options.rest;
+    if (rest) {
+      this.logger.debug('deploy with REST');
+      const headers = {
+        Authorization: this && `OAuth ${this.accessToken}`,
+        clientId: this.oauth2 && this.oauth2.clientId,
+        'Sforce-Call-Options': 'client=sfdx-core',
+      };
+      const url = `${this.baseUrl()}/metadata/deployRequest`;
+      const request = Transport.prototype._getHttpRequestModule();
```

Comment:
Ugg, we should keep this in mind with jsforce v2. I know we want this in jsforce but it would be nice to make this better for people trying to extend jsforce.

Reply:
yeah, this and the above `Buffer` to `fs.ReadStream` are the two big things I don't like about how we're doing it. I wasn't sure what the best way to do the request was here